docs - rework the host/port topic, add section for listen address

lisakowen opened the pull request:
add some hardening info based on https://github.com/greenplum-db/pxf/issues/974 to the existing pxf docs.  note that this info will be further updated in a separate PR to address the introduction of server.address to pxf-application.properties in https://github.com/greenplum-db/pxf/pull/976.

doc review site link (behind vpn):  https://docs-staging.vmware.com/en/VMware-Tanzu-Greenplum-Platform-Extension-Framework/tmpbranch/tanzu-greenplum-platform-extension-framework/GUID-cfghostport.html

denalex commented on docs/content/cfghostport.html.md.erb:
```diff
@@ -1,21 +1,93 @@
+The `server.address` property identifies the IP address or hostname of the network interface on which the PXF service listens. The default PXF service listen address is `0.0.0.0`.
 
-Perform the following procedure to configure the PXF Service host and/or port number on one or more Greenplum Database hosts:
+To secure communications and restrict PXF to listen only on the loopback interface, set the `server.address` to `localhost`:
```
we need to mention that this can be done only if PXF Service JVM runs on Greenplum hosts.

denalex commented on docs/content/cfghostport.html.md.erb:
````diff
@@ -1,21 +1,93 @@
+
+    ``` shell
+    gpadmin@gpmaster$ pxf cluster restart
+    ```
````
this is a bit tricky for the master, since the update to `~/.bashrc` with the new PXF_PORT value will not be immediately effective unless the operator does `source ~/.bashrc`. Without doing this, PXF on master might still start on the old port (need to be tested, though).

Also, if a user sets `pxf.port` variable in `pxf-application.properties` it will take an effect without requiring PXF_PORT env variable (for Java process only, PXF_PORT still needs to be set for GP to find PXF).

lisakowen replied:
thank you for reviewing, @denalex .

i added a step for the user to source the .bashrc they just updated.

we do not currently document pxf.port.  and it doesn't appear to be in the installed pxf-application.properties file.  do we still want to mention it?

denalex commented on docs/content/cfghostport.html.md.erb:
```diff
@@ -1,21 +1,93 @@
+
+## <a id="host"></a>Configuring the Host
+
+<div class="note"><b>Note:</b> You must restart both Greenplum Database and PXF when you configure the host in this manner. Consider performing this configuration during a scheduled down time.</div>
```
we should describe here that this is a totally different deployment topology where PXF JVM is not located on the same hosts as GP. In fact, I think the deployment topologies should be a separate topic where we describe how to achieve this, not a side-note here (use case driven vs. config driven).

PXF will not need to get restarted, since PXF JVMs will be located elsewhere. In fact, none of the `pxf cluster` commands will be working since that command assumes PXF JVMs are on GP hosts.

lisakowen replied:
hi @denalex , circling back on this PR.  we never explicitly documented the current deployment topology (pxf service located on greenplum hosts) nor the alternate deployment topology (pxf service not located on greenplum hosts).  i'll pull together a topic for this.

denalex commented on docs/content/cfghostport.html.md.erb:
````diff
@@ -25,24 +97,27 @@ Perform the following procedure to configure the PXF Service host and/or port nu
+
+    ``` shell
+    gpadmin@gpmaster$ pxf cluster restart
+    ```
````
not needed and will not work

lisakowen commented on docs/content/cfghostport.html.md.erb:
```diff
@@ -1,21 +1,99 @@
 ---
 
-By default, a PXF Service started on a Greenplum host listens on port number `5888` on `localhost`. You can configure PXF to start on a different port number, or use a different hostname or IP address. To change the default configuration, you will set one or both of the environment variables identified below:
+By default, the PXF Service starts on a Greenplum host and listens on `0.0.0.0:5888`. With this configuration, the PXF Service listens on all network interfaces of the Greenplum host and accepts network requests from anywhere. You can configure PXF to listen on a different local address. You can also configure PXF to listen on a different port number, or to run on a different host. To change the  ...
```
thanks for taking a look at this PR, @ashuka24 .  these changes are for the *current docs* (v6.6.0).  the v6.7.0 updates will include the server.address default change to localhost.

ashuka24 replied:
ah ok. I misunderstood the PR comment. Thanks for the clarification.

yimingli-vmware commented on docs/content/cfghostport.html.md.erb:
```diff
@@ -25,24 +103,27 @@ Perform the following procedure to configure the PXF Service host and/or port nu
-    1. Identify the host name or IP address of the PXF Service.
-    2. Identify the port number on which you want the PXF Service to run.
+    1. Identify the host name or IP address of a PXF Server.
     3. Log in to the Greenplum Database host:
```
the numbers haven't been updated after this line

lisakowen commented:
reminder:  this PR is for the current (6.6.0) docs.  there will be further revisions to this topic for new 6.7.0 features/changes.

i think i've addressed the outstanding comments.  i added a new topic about the pxf deployment topology at the topmost level.  (i can add it lower in the subnav if desired, like under Administering.)  the topic calls out the basic info for the default and alternate topologies, is kind of thin on content.  i also updated the "configuring the host" topic.

aside from this new content, all of the docs are written for the default deployment topology.  i have not reviewed all topics to call out changes related to alternate topology - should i?  for example, most admin procedures use the "pxf cluster xxx" commands, which aren't applicable to the alternate topology (though i do call this out in the new deployment topology topic).  the config page and some of other doc topics assert that the pxf service is running on greenplum hosts.  thoughts?

denalex commented on docs/content/cfghostport.html.md.erb:
```diff
@@ -25,24 +58,74 @@ Perform the following procedure to configure the PXF Service host and/or port nu
+
+If you have chosen the [alternate deployment topology](deployment_topos.html#alt_topo) for PXF, you must set the `PXF_HOST` environment variable on each Greenplum segment host to inform Greenplum of the location of the PXF service.
+
+Perform the following procedure to configure the PXF host on each Greenplum Database segmet host:
```
```suggestion
Perform the following procedure to configure the PXF host on each Greenplum Database segment host:
```

lisakowen commented:
i am going to merge this (for v6.6.0 docs), so i can submit the follow-on PR for 6.7.0.  as always, still accepting comments if others wish to review.
